### ISSUE TITLE
Rspec Issue:  Missing locales in one spec breaks another

### ISSUE THREAD

**Issue by lynnfaraday** [svenfuchs/i18n#173]
I'm having a weird error with the rspec tests in my app related to the I18n library.   This is a Ruby app, but not Rails.   I18n is version 0.6.1

If one of my tests calls I18n.t _without first setting the load_path_, it appears to somehow mess up subsequent tests that _do_ actually set the load path.   To illustrate this failure in a very simple manner, I created two specs that don't even use any application code:

```
(just showing the specs for brevity; can provide the full source on request)
--test1.rb--
 it "should fail if a translation is not found" do
      puts "Translation fail"
      I18n.t('test').should eq "translation missing: en.test"
 end

--test2.rb--
it "should translate hello to english" do
  create_locale file_for_testing() 
  I18n.load_path << Dir.pwd + "/en.yml"
  puts "Translate English"
  I18n.locale = :en
  I18n.t("hello").should eq "Hello world"
end
```

If I adjust the file names so that the "English" translation test runs first, everything's fine:

```
Translate English
.Translation fail 
...
2 examples, 0 failures
```

If I adjust the file names such that the "fail" translation test runs first, it causes the "English" test to fail too!  

```
Translation fail
.Translate English
F

Failures:

  1) foo should translate hello to english
     Failure/Error: I18n.t("hello").should eq "Hello world"

       expected: "Hello world"
            got: "translation missing: en.hello"
   ...
    2 examples, 1 failure
```

Clearly the first spec is somehow messing things up for the second spec.  But what?  And why?

**Comment by radar** [comment]
Hi there,

It's been a while, but I don't suppose anyone has an app which reproduces this issue? That'd help tremendously in debugging this.

Thanks!

**Comment by lynnfaraday** [comment]
@radar    It's not an app, but here's a [gist](https://gist.github.com/lynnfaraday/c64e2e4d9a3c91a8eac4592ff95b4968) that illustrates the issue.

Just create the three files and run rspec.

`rspec spec.rb` works - both tests pass

`rspec spec_failed.rb` illustrates the problem - both tests fail

The only difference between the two spec files is the order of the tests.

**Comment by radar** [comment]
Thanks @lynnfaraday!

I've been able to investigate this today and I've discovered a solution. However, it doesn't work with `I18n.load_path <<`, only `I18n.load_path +=`. I think to make it work with the former we'd need to re-define the `<<` method on `load_path`, which may be a bit hacky. tbh I'd rather advise people to use `I18n.load_path +=`.

**Comment by radar** [comment]
☝️  there's a PR which will fix your issue. Here's an updated Gist of the code I used to verify this patch: https://gist.github.com/radar/c74cadff6dee7988f764536d667f44df

**Comment by radar** [comment]
Just merged #348 which fixes this. Thanks @lynnfaraday!
